docs/examples/dvs-from-file.ipynb

```diff
@@ -0,0 +1,318 @@
+    "    t0_us = t_frame\n",
+    "    t1_us = t_frame + dt_frame_us\n",
+    "    t = dvs_events.events[:][\"t\"]\n",
+    "    m = (t >= t0_us) & (t < t1_us)\n",
+    "    events_m = dvs_events.events[m]\n",
+    "\n",
+    "    # show \"off\" (0) events as -1 and \"on\" (1) events as +1\n",
+    "    events_sign = 2.0 * events_m[\"p\"] - 1\n",
+    "\n",
+    "    frame_img = np.zeros((dvs_height, dvs_width))\n",
+    "    frame_img[events_m[\"y\"], events_m[\"x\"]] = events_sign\n",
+    "\n",
+    "    img = plt.imshow(frame_img, vmin=-1, vmax=1, animated=True)\n",
+    "    imgs.append([img])\n",
+    "\n",
+    "del dvs_events\n",
+    "\n",
+    "ani = ArtistAnimation(fig, imgs, interval=50, blit=True)\n",
+    "HTML(ani.to_html5_video())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Nengo network\n",
+    "\n",
+    "We can now load our data into a NengoLoihi network, using the\n",
+    "`nengo_loihi.DVSFileChipNode` type of `nengo.Node`. We pass the node the path to our\n",
+    "data file. We also use the optional `pool` parameter to pool over space. This allows our\n",
+    "node to output at a lower spatial resolution than the standard 180 x 240 pixel DVS\n",
+    "output. We then make two ensembles: one to receive the positive polarity \"on\" events,\n",
+    "and the other to receive the negative polarity \"off\" events. Since we passed\n",
+    "`channels_last=True` to our node, the polarity will be the least-significant index of\n",
+    "our node output, and we connect it to the ensembles accordingly (by sending all even\n",
+    "outputs to the negative polarity ensemble `ensembles[0]`, and the odd outputs to the\n",
+    "positive polarity ensemble `ensembles[1]`). We run the simulation and collect the spikes\n",
+    "from both the negative and positive polarity ensembles."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pool = (10, 10)\n",
+    "\n",
+    "height = 180 // pool[0]\n",
+    "width = 240 // pool[1]\n",
+    "gain = 101\n",
+    "\n",
+    "with nengo.Network() as net:\n",
+    "    u = nengo_loihi.DVSFileChipNode(\n",
+    "        file_path=events_file_name, pool=pool, channels_last=True\n",
+    "    )\n",
+    "    assert u.height == height and u.width == width\n",
+    "    assert u.polarity == 2\n",
+    "\n",
+    "    ensembles = [\n",
+    "        nengo.Ensemble(\n",
+    "            height * width,\n",
+    "            1,\n",
+    "            neuron_type=nengo.SpikingRectifiedLinear(),\n",
+    "            gain=nengo.dists.Choice([gain]),\n",
+    "            bias=nengo.dists.Choice([0]),\n",
+    "        )\n",
+    "        for _ in range(u.polarity)\n",
+    "    ]\n",
+    "\n",
+    "    for k, e in enumerate(ensembles):\n",
+    "        u_channel = u[k :: u.polarity]\n",
+    "        nengo.Connection(u_channel, e.neurons, transform=1.0 / np.prod(pool))\n",
```

Comment:
We could. I would want things to be consistent with the live DVS (when we add that), and there it might be a bit harder to do that. The way that live DVS works is you basically have one neuron per input pixel that gets the input from the DVS and relays the spikes; you can then arbitrarily control the axons coming out of those neurons to direct the DVS input. So for pooling, I just have multiple axons go to the same target synapses. To do the scaling, there might be a bit of work necessary in the builder to find which synapses things are being redirected to, and scale them accordingly. Probably not terribly difficult, but certainly a bit more work.

The scaling would certainly be more like average/max pooling if we had the scaling factor in there, and people who are coming from an ML background would be more familiar with those.

Reply:
I don't think it's that critical, so if it makes the implementation simpler to leave it out I'd go with that. Maybe just adding a comment in the notebook explaining why scaling is being applied, to make it stand out a bit more for anyone reading the example (so they'll think about whether they want to be adding that in their own code).